tola/tests/test_util.py

```diff
@@ -245,3 +249,131 @@ def test_save_data_to_silo_skipped_rows(self):
 
         result = save_data_to_silo(self.silo, data, self.read)
         self.assertEqual(result, expected_response)
+
+    @patch('logging.warning')
+    def test_ona_parse_type_group_with_valid_data(self, mock_logger):
+        silo = factories.Silo(name='test_ona')
+        read = factories.Read(read_name='test_ona_read',
+                              owner=self.tola_user.user)
+        silo.reads.add(read)
+        data = [{
+                    '_notes': [],
+                    'Name': 'Bob',
+                    'Number': '1',
+                    '_edited': False,
+                    '_status': 'submitted_via_web',
+                    '_submission_time': '2018-06-26T15:01:19',
+                    '_id': 31108803
+                }, {
+                    '_notes': [],
+                    'Name': 'Jason',
+                    'Number': '2',
+                    '_edited': False,
+                    '_status': 'submitted_via_web',
+                    '_submission_time': '2018-06-26T15:01:19',
+                    '_id': 31108806
+                }]
+
+        form_data = [{
+                        'bind': {
+                            'required': 'true'
+                        },
+                        'type': 'text',
+                        'name': 'Name',
+                        'label': 'Name'
+                    }, {
+                        'bind': {
+                            'required': 'true'
+                        },
+                        'type': 'text',
+                        'name': 'Number',
+                        'label': 'Number'
+                    }, {
+                        'bind': {
+                            'calculate': "'vviKGm8pxer5Zu3E3H3wbf'"
+                        },
+                        'type': 'calculate',
+                        'name': '__version__'
+                    }, {
+                        'control': {
+                            'bodyless': True
+                        },
+                        'type': 'group',
+                        'children': [{
+                            'bind': {
+                                'readonly': 'true()',
+                                'calculate': "concat('uuid:', uuid())"
+                            },
+                            'type': 'calculate',
+                            'name': 'instanceID'
+                        }],
+                        'name': 'meta'
+                    }]
+
+        with mock.patch.object(logger, 'warning') as mock_logger:
+            ona_parse_type_group(data, form_data, '', silo, read)
+            mock_logger.assert_not_called()
+
+    def test_ona_parse_type_group_with_invalid_keys(self):
```

Comment:
It's good to mock the function `ona_parse_type_group`. It's called in the function. `ona_parse_type_group`